.github/workflows/deploy.yaml

```diff
@@ -3,16 +3,51 @@ on:
     branches:
       - master
       - develop
-
+      - feature/bump-pubspec-version-automatically
 env:
   JAVA_VERSION: "11.x"
   FLUTTER_VERSION: "2.0.1"
   PROPERTIES_PATH: "android/key.properties"
 
 jobs:
+  bump_version:
+    name: "Bump pubspec version"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./app_feup
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.NIAEFEUPBOT_PAT }}
+          fetch-depth: 2
+
+      - name: Ensure app_version was not changed manually
+        run: git diff HEAD^:app_version.txt HEAD:app_version.txt
```

Comment:
Won't this just test against the previous commit? This should apply test against develop OR master